### PR TITLE
object_zl1

### PR DESCRIPTION
**Pull request by hack-nuss**
OOT documentation applied

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: assets/xml/objects/object_zl1.xml [hunk 1]
```diff
@@ -1,90 +1,114 @@
...
-        <Animation Name="object_zl1_Anim_011B88" Offset="0x11B88" />
-        <Animation Name="object_zl1_Anim_012118" Offset="0x12118" />
-        <Animation Name="object_zl1_Anim_012B04" Offset="0x12B04" />
-        <Animation Name="object_zl1_Anim_012F80" Offset="0x12F80" />
-        <Animation Name="object_zl1_Anim_0132D8" Offset="0x132D8" />
-        <Animation Name="object_zl1_Anim_0138E0" Offset="0x138E0" />
-        <Animation Name="object_zl1_Anim_013F10" Offset="0x13F10" />
-        <Animation Name="object_zl1_Anim_0143A8" Offset="0x143A8" />
+        <!-- Child Zelda 1 Skeleton -->
+        <Skeleton Name="gChildZelda1Skel" Type="Flex" LimbType="Standard" LimbNone="OBJECT_ZL1_LIMB_NONE" LimbMax="OBJECT_ZL1_LIMB_MAX" EnumName="ObjectZl1Limb" Offset="0xF5D8" />
```
Comment (hensldm) [inline assets/xml/objects/object_zl1.xml:4]:
```suggestion
        <Skeleton Name="gChildZelda1Skel" Type="Flex" LimbType="Standard" LimbNone="CHILD_ZELDA_1_LIMB_NONE" LimbMax="CHILD_ZELDA_1_LIMB_MAX" EnumName="ChildZelda1Limb" Offset="0xF5D8" />
```

File: assets/xml/objects/object_zl1.xml [hunk 2]
```diff
@@ -1,90 +1,114 @@
 ﻿<Root>
     <File Name="object_zl1" Segment="6">
```
Comment (hensldm) [inline assets/xml/objects/object_zl1.xml:2]:
```suggestion
﻿<Root>
﻿    <!-- Object for Child Zelda 1 -->
    <File Name="object_zl1" Segment="6">
```

**Review by hensldm — CHANGES_REQUESTED** [review]
Could you sort these back in Offset order as well?

**Review by hensldm — COMMENTED** [review]
Offsets still seem out of order

File: assets/xml/objects/object_zl1.xml (hunk 2 above)
Comment (isghj5) [reply assets/xml/objects/object_zl1.xml]: Except... this is the unused object, It gets mentioned in the empty actor ovl_En_Zl1, but that actor doesn't use the object at all with an empty draw function and no references to the contents of this object.

Should we include "Unused" in the description comment?

File: assets/xml/objects/object_zl1.xml [hunk 3]
```diff
@@ -1,90 +1,115 @@
...
+        <Texture Name="gChildZelda1HeaddressTriforceEmblemTex" OutName="child_zelda_1_headdress_triforce_emblem" Format="rgba16" Width="16" Height="32" Offset="0x7AC8" />
+        <Texture Name="gChildZelda1PointedHairOnSideTex" OutName="child_zelda_1_pointed_hair_on_side" Format="rgba16" Width="16" Height="16" Offset="0x7EC8" />
+        <Texture Name="gChildZelda1HairTex" OutName="child_zelda_1_hair" Format="rgba16" Width="16" Height="8" Offset="0x80C8" />
+        <Texture Name="gChildZelda1Tex_81C8" OutName="child_zelda_1_tex_81C8" Format="ci8" Width="8" Height="8" Offset="0x81C8" TlutOffset="0x70C8" />
+        <Texture Name="gChildZelda1EarTex" OutName="child_zelda_1_ear" Format="rgba16" Width="16" Height="16" Offset="0x8208" />
+        <Texture Name="gChildZelda1AlternateDressPatternTex" OutName="child_zelda_1_alternate_dress_pattern" Format="ci8" Width="8" Height="8" Offset="0x8408" TlutOffset="0x70C8" />
+        <Texture Name="gChildZelda1ALternateDressBeltTex" OutName="child_zelda_1_alternate_dress_belt" Format="ci8" Width="32" Height="8" Offset="0x8448" TlutOffset="0x70C8" />
```
Comment (engineer124) [inline assets/xml/objects/object_zl1.xml:42]:
```suggestion
        <Texture Name="gChildZelda1AlternateDressBeltTex" OutName="child_zelda_1_alternate_dress_belt" Format="ci8" Width="32" Height="8" Offset="0x8448" TlutOffset="0x70C8" />
```

File: assets/xml/objects/object_zl1.xml [hunk 4]
```diff
@@ -1,90 +1,115 @@
...
+        <Texture Name="gChildZelda1EyeOpenLookingRightTex" OutName="child_zelda_1_eye_open_looking_right" Format="ci8" Width="32" Height="32" Offset="0x8C48" TlutOffset="0x6CD8" />
+        <Texture Name="gChildZelda1EyeHalf1Tex" OutName="child_zelda_1_eye_half_1" Format="ci8" Width="32" Height="32" Offset="0x9448" TlutOffset="0x6CD8" />
+        <Texture Name="gChildZelda1EyeHalf2Tex" OutName="child_zelda_1_eye_half_2" Format="ci8" Width="32" Height="32" Offset="0x9848" TlutOffset="0x6CD8" />
+        <Texture Name="gChildZelda1EyeClosedTex" OutName="child_zelda_1_eye_closed" Format="ci8" Width="32" Height="32" Offset="0x9C48" TlutOffset="0x6CD8" />
+        <Texture Name="gChildZelda1EyeWideTex" OutName="child_zelda_1_eye_wide" Format="ci8" Width="32" Height="32" Offset="0xA048" TlutOffset="0x6CD8" />
+
+        <!-- Palettes -->
+        <Texture Name="gChildZelda1TLUT_0450" OutName="child_zelda_1_tlut_0450" Format="rgba16" Width="16" Height="16" Offset="0x0450" />
+        <Texture Name="gChildZelda1TLUT_6CD8" OutName="child_zelda_1_tlut_6CD8" Format="rgba16" Width="252" Height="1" Offset="0x6CD8" />
```
Comment (engineer124) [inline assets/xml/objects/object_zl1.xml:15]: Offsets are still very out of order

File: assets/xml/objects/object_zl1.xml [hunk 5]
```diff
@@ -1,90 +1,119 @@
...
-        <Animation Name="object_zl1_Anim_012F80" Offset="0x12F80" />
-        <Animation Name="object_zl1_Anim_0132D8" Offset="0x132D8" />
-        <Animation Name="object_zl1_Anim_0138E0" Offset="0x138E0" />
-        <Animation Name="object_zl1_Anim_013F10" Offset="0x13F10" />
-        <Animation Name="object_zl1_Anim_0143A8" Offset="0x143A8" />
+        <!-- Palette -->
+        <Texture Name="gChildZelda1TLUT_0450" OutName="child_zelda_1_tlut_0450" Format="rgba16" Width="16" Height="16" Offset="0x0450" />
+
+        <!-- Textures -->
+        <Texture Name="gChildZelda1DressGradientTex" OutName="child_zelda_1_dress_gradient" Format="ci8" Width="8" Height="8" Offset="0x0650" TlutOffset="0x0450" />
```
Comment (engineer124) [inline assets/xml/objects/object_zl1.xml:8]:
```suggestion
        <Texture Name="gChildZelda1TLUT_0450" OutName="child_zelda_1_tlut_0450" Format="rgba16" Width="16" Height="16" Offset="0x450" />

        <!-- Textures -->
        <Texture Name="gChildZelda1DressGradientTex" OutName="child_zelda_1_dress_gradient" Format="ci8" Width="8" Height="8" Offset="0x650" TlutOffset="0x450" />
```
We've put effort to remove leading zero's, could you double check leading 0's in the offsets/tlutOffsets?

File: assets/xml/objects/object_zl1.xml [hunk 6]
```diff
@@ -1,90 +1,119 @@
...
+        <Texture Name="gChildZelda1EyeOpenLookingLeftTex" OutName="child_zelda_1_eye_open_looking_left" Format="ci8" Width="32" Height="32" Offset="0x8848" TlutOffset="0x6CD8" />
+        <Texture Name="gChildZelda1EyeOpenLookingRightTex" OutName="child_zelda_1_eye_open_looking_right" Format="ci8" Width="32" Height="32" Offset="0x8C48" TlutOffset="0x6CD8" />
+        <Texture Name="gChildZelda1EyeHalf1Tex" OutName="child_zelda_1_eye_half_1" Format="ci8" Width="32" Height="32" Offset="0x9448" TlutOffset="0x6CD8" />
+        <Texture Name="gChildZelda1EyeHalf2Tex" OutName="child_zelda_1_eye_half_2" Format="ci8" Width="32" Height="32" Offset="0x9848" TlutOffset="0x6CD8" />
+        <Texture Name="gChildZelda1EyeClosedTex" OutName="child_zelda_1_eye_closed" Format="ci8" Width="32" Height="32" Offset="0x9C48" TlutOffset="0x6CD8" />
+        <Texture Name="gChildZelda1EyeWideTex" OutName="child_zelda_1_eye_wide" Format="ci8" Width="32" Height="32" Offset="0xA048" TlutOffset="0x6CD8" />
+
+        <!-- Palettes -->
+        <Texture Name="gChildZelda1TLUT_6CD8" OutName="child_zelda_1_tlut_6CD8" Format="rgba16" Width="252" Height="1" Offset="0x6CD8" />
```
Comment (engineer124) [inline assets/xml/objects/object_zl1.xml:64]: Offsets are still out of order. It jumps from `0xA048` down to `0x6CD8`

**Review by hensldm — APPROVED** [review]

File: assets/xml/objects/object_zl1.xml [hunk 7]
```diff
@@ -1,90 +1,115 @@
...
-        <Animation Name="object_zl1_Anim_011B88" Offset="0x11B88" />
-        <Animation Name="object_zl1_Anim_012118" Offset="0x12118" />
-        <Animation Name="object_zl1_Anim_012B04" Offset="0x12B04" />
-        <Animation Name="object_zl1_Anim_012F80" Offset="0x12F80" />
-        <Animation Name="object_zl1_Anim_0132D8" Offset="0x132D8" />
-        <Animation Name="object_zl1_Anim_0138E0" Offset="0x138E0" />
-        <Animation Name="object_zl1_Anim_013F10" Offset="0x13F10" />
-        <Animation Name="object_zl1_Anim_0143A8" Offset="0x143A8" />
+        <!-- Animation -->
+        <Animation Name="gChildZelda1Anim_00438" Offset="0x0438" />
```
Comment (engineer124) [inline assets/xml/objects/object_zl1.xml:5]:
```suggestion
        <Animation Name="gChildZelda1Anim_00438" Offset="0x438" />
```

File: assets/xml/objects/object_zl1.xml [hunk 8]
```diff
@@ -1,90 +1,115 @@
...
-        <DList Name="object_zl1_DL_005CD8" Offset="0x5CD8" />
-        <DList Name="object_zl1_DL_005E28" Offset="0x5E28" />
-        <DList Name="object_zl1_DL_006008" Offset="0x6008" />
-        <DList Name="object_zl1_DL_006828" Offset="0x6828" />
-        <DList Name="object_zl1_DL_0069E0" Offset="0x69E0" />
-        <DList Name="object_zl1_DL_006BA0" Offset="0x6BA0" />
-        <!-- <Texture Name="object_zl1_Tex_006CD8" OutName="tex_006CD8" Format="rgba16" Width="16" Height="16" Offset="0x6CD8" /> -->
-        <!-- <Blob Name="object_zl1_Blob_006ED8" Size="0x1F0" Offset="0x6ED8" /> -->
-        <!-- <Texture Name="object_zl1_TLUT_0070C8" OutName="tlut_0070C8" Format="rgba16" Width="16" Height="16" Offset="0x70C8" /> -->
-        <!-- <Blob Name="object_zl1_Blob_0072C8" Size="0x740" Offset="0x72C8" /> -->
```
Comment (engineer124) [inline assets/xml/objects/object_zl1.xml:33]: A lot of blobs seems to have been lost. Could you bring these comments back? There's more below as well

File: assets/xml/objects/object_zl1.xml [hunk 9]
```diff
@@ -1,90 +1,122 @@
...
+        <DList Name="gChildZelda1LeftHandDL" Offset="0x5E28" />
+        <DList Name="gChildZelda1WaistDL" Offset="0x6008" />
+        <DList Name="gChildZelda1Dress1DL" Offset="0x6828" />
+        <DList Name="gChildZelda1Dress2DL" Offset="0x69E0" />
+        <DList Name="gChildZelda1Dress3DL" Offset="0x6BA0" />
+
+
+        <!-- Textures -->
         <!-- <Texture Name="object_zl1_Tex_006CD8" OutName="tex_006CD8" Format="rgba16" Width="16" Height="16" Offset="0x6CD8" /> -->
+        <Texture Name="gChildZelda1TLUT_6CD8" OutName="child_zelda_1_tlut_6CD8" Format="rgba16" Width="252" Height="1" Offset="0x6CD8" />
```
Comment (AngheloAlf) [inline assets/xml/objects/object_zl1.xml:42]: This two entries are the same
```suggestion
        <Texture Name="gChildZelda1TLUT_6CD8" OutName="child_zelda_1_tlut_6CD8" Format="rgba16" Width="252" Height="1" Offset="0x6CD8" />
```

File: assets/xml/objects/object_zl1.xml [hunk 10]
```diff
@@ -1,90 +1,122 @@
...
         <!-- <Blob Name="object_zl1_Blob_006ED8" Size="0x1F0" Offset="0x6ED8" /> -->
+        <Texture Name="gChildZelda1TLUT_6ED0" OutName="child_zelda_1_tlut_6ED0" Format="rgba16" Width="252" Height="1" Offset="0x6ED0" />
         <!-- <Texture Name="object_zl1_TLUT_0070C8" OutName="tlut_0070C8" Format="rgba16" Width="16" Height="16" Offset="0x70C8" /> -->
+        <Texture Name="gChildZelda1TLUT_70C8" OutName="child_zelda_1_tlut_70C8" Format="rgba16" Width="160" Height="1" Offset="0x70C8" />
```
Comment (AngheloAlf) [inline assets/xml/objects/object_zl1.xml:46]: Same here
```suggestion
        <Texture Name="gChildZelda1TLUT_6ED0" OutName="child_zelda_1_tlut_6ED0" Format="rgba16" Width="252" Height="1" Offset="0x6ED0" />
        <Texture Name="gChildZelda1TLUT_70C8" OutName="child_zelda_1_tlut_70C8" Format="rgba16" Width="160" Height="1" Offset="0x70C8" />
```

File: assets/xml/objects/object_zl1.xml [hunk 11]
```diff
@@ -1,90 +1,122 @@
...
         <!-- <Blob Name="object_zl1_Blob_0072C8" Size="0x740" Offset="0x72C8" /> -->
```
Comment (AngheloAlf) [inline assets/xml/objects/object_zl1.xml:47]: This is out of order

File: assets/xml/objects/object_zl1.xml [hunk 12]
```diff
@@ -1,90 +1,122 @@
...
+        <Texture Name="gChildZelda1AlternateDressLiningTex" OutName="child_zelda_1_alternate_dress_lining" Format="rgba16" Width="8" Height="16" Offset="0x8648" />
+        <Texture Name="gChildZelda1BootTex" OutName="child_zelda_1_boot" Format="rgba16" Width="8" Height="8" Offset="0x8748" />
+        <Texture Name="gChildZelda1BootShaftTex" OutName="child_zelda_1_boot_shaft" Format="rgba16" Width="8" Height="8" Offset="0x87C8" />
         <!-- <Blob Name="object_zl1_Blob_008848" Size="0x2000" Offset="0x8848" /> -->
-        <Texture Name="object_zl1_Tex_00A848" OutName="tex_00A848" Format="rgba16" Width="8" Height="8" Offset="0xA848" />
-        <Texture Name="object_zl1_Tex_00A8C8" OutName="tex_00A8C8" Format="ci8" Width="8" Height="16" Offset="0xA8C8" />
-        <Texture Name="object_zl1_Tex_00A948" OutName="tex_00A948" Format="ci8" Width="16" Height="32" Offset="0xA948" />
+        <Texture Name="gChildZelda1EyeOpenLookingLeftTex" OutName="child_zelda_1_eye_open_looking_left" Format="ci8" Width="32" Height="32" Offset="0x8848" TlutOffset="0x6CD8" />
```
Comment (AngheloAlf) [inline assets/xml/objects/object_zl1.xml:64]: This two refer to the same offset  too

**Review by engineer124 — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**